cmd/image-autobumper/updater/updater.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package updater handles creation and updates of GitHub PullRequests.
+package updater
+
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/test-infra/prow/github"
+)
+
+// Indicates whether maintainers can modify a pull request in fork.
+const (
+	AllowMods   = true
+	PreventMods = false
+)
+
+type updateClient interface {
+	UpdatePullRequest(org, repo string, number int, title, body *string, open *bool, branch *string, canModify *bool) error
+	BotUser() (*github.UserData, error)
+	FindIssues(query, sort string, asc bool) ([]github.Issue, error)
+}
+
+type ensureClient interface {
+	updateClient
+	AddLabel(org, repo string, number int, label string) error
+	CreatePullRequest(org, repo, title, body, head, base string, canModify bool) (int, error)
+	GetIssue(org, repo string, number int) (*github.Issue, error)
+}
+
+func EnsurePRWithQueryTokens(org, repo, title, body, source, baseBranch, queryTokensString string, allowMods bool, gc ensureClient) (*int, error) {
+	n, err := updatePRWithQueryTokens(org, repo, title, body, queryTokensString, gc)
```

Comment:
What is the meaning of `n`? It not easy to infer.

Reply:
No reusable PRs found. Correct?

Reply:
Pointer to PR number. Got it. But proper naming could help the reader.

Reply:
reworked.